File: python-code/ZooKeeper.py
#!/usr/bin/env python
# coding: utf-8

# In[3]:

from Animals import Animals
from abc import *
# from zooAnouncer import zooAnouncer

class zooInterface():
    def registerObserver(self): pass
    def removeObserver(self): pass
    def notifyObserver(self): pass

class ZooKeeper(zooInterface):

    def __init__(self):
        self.observers = []
    #     Zoo keeper performs his duty through different methods listed below.
#     Each method takes it animal object as a parameter so it can ask corresponding move for animals to behave
#     when Zoo Keeper executes certain action.

    def registerObserver(self, observer):
        self.observers.append(observer)
        # print(observer)
    def removeObserver(self, observer):
        self.observers.remove(observer)
    def notifyObserver(self, order):
        for obs in self.observers:
            obs.update(order)

    def wake(self, Animals):
        self.notifyObserver("wake animals")
        Animals.wakingUp()

    def rollCall(self, Animals):
        self.notifyObserver("Calling animals")
        Animals.makeNoise()

    def feed(self, Animals):
        self.notifyObserver("feed animals")
        Animals.eat()

    def exercise(self, Animals):
        self.notifyObserver("exercise animals")
        Animals.roam()

    def shut(self, Animals):
        self.notifyObserver("Shut the zoo")
        Animals.sleep()

File: project 1/src/zoo/ZooKeeper.java
package zoo;
public class ZooKeeper {
	/* Zoo keeper performs his duty through different methods listed below.
	 * Each method takes it animal object as a parameter so it can ask corresponding move for animals to behave
	 * when Zoo Keeper executes certain action.
	 * 
	*/
	public void wake(Animals animal) {
		System.out.println("wake animals");
		animal.wakingUp();
	}
	
	public void rollCall(Animals animal) {
		System.out.println("Calling animals");
		animal.makeNoise();
	}
	
	public void feed(Animals animal) {
		System.out.println("feed animals");
		animal.eat();
	}
	
	public void exercise(Animals animal) {
		System.out.println("exercise animals");
		animal.roam();
	}
	
	public void shut(Animals animal) {
		System.out.println("Shut the zoo");
		animal.sleep();
	}
}
